FusionIIIT/applications/hostel_management/models.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+from django.db import models
+from django.contrib.auth.models import User
+from applications.globals.models import ExtraInfo, Staff
+from applications.academic_information.models import Student
+from applications.complaint_system.models import Caretaker
+from django.utils import timezone
+
+
+class HostelManagementConstants:
+    ROOM_STATUS = (
+        ('Booked', 'Booked'),
+        ('CheckedIn', 'CheckedIn'),
+        ('Available', 'Available'),
+        ('UnderMaintenance', 'UnderMaintenance'),
+        )
+
+    DAYS_OF_WEEK = (
+            (0, 'Monday'),
+            (1, 'Tuesday'),
+            (2, 'Wednesday'),
+            (3, 'Thursday'),
+            (4, 'Friday'),
+            (5, 'Saturday'),
+            (6, 'Sunday')
+        )
+
+    BOOKING_STATUS = (
+    ("Confirmed" , 'Confirmed'),
+    ("Pending" , 'Pending'),
+    ("Rejected" , 'Rejected'),
+    ("Canceled" , 'Canceled'),
+    ("CancelRequested" , 'CancelRequested'),
+    ("CheckedIn" , 'CheckedIn'),
+    ("Complete", 'Complete'),
+    ("Forward", 'Forward')
+    )
+
+    HALL_NO = (
+        ('HALL-1','Hall-1'),
+        ('HALL-3','Hall-3'),
+        ('HALL-4','Hall-4'),
+    )
+
+
+class Hall(models.Model):
+    hall_no = models.CharField(max_length=15, choices=HostelManagementConstants.HALL_NO, default='')
+    max_accomodation = models.IntegerField(default=0)
+    number_students = models.PositiveIntegerField(default=0)
+    hall_caretaker = models.ForeignKey(Caretaker, on_delete=models.CASCADE)
```

Comment:
There can be multiple caretakers for a hall. So, it should be a ManyToManyField.

Reply:
yes its done.